Extension/src/LanguageServer/ui_new.ts

```diff
@@ -280,10 +280,10 @@ export class NewUI implements UI {
         const settings: CppSettings = new CppSettings((vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) ? vscode.workspace.workspaceFolders[0]?.uri : undefined);
 
         // TODO: Integrate with Tarik's feature to determine if compiler/bare-intellisense is configured
-        if (settings.intelliSenseEngine === "disabled") {
+        if (settings.intelliSenseEngine === "default" && !compilerDefaults.trustedCompilerFound) {
```

Comment:
I think we also need to check like `(compilerPaths.length !== 1 || compilerPaths[0] !== "")` i.e. in case compilerPath is set to `""` then we don't want to do this.